### PR TITLE
Tools and interfaces for boost+sse+webglcanvas

### PR DESCRIPTION
**Pull request by afustePTC**
- Pathpoint, path and motion tools
- Pathpoint, path and motion nodes
- Lego Boost SSE

### CODE REVIEW

File: interfaces/lego-boost-sse/tools/mission/index.html [hunk 1]
```diff
@@ -0,0 +1,284 @@
+<!doctype html>
```
Comment (hobinjk-ptc) [inline interfaces/lego-boost-sse/tools/mission/index.html:1]: Is this file doing something here? I wasn't aware that there could be a tools directory inside a hardware interface

**Review by hobinjk-ptc — APPROVED** [review]
It would be nice to delete the random .DS_Store files added in this PR but overall this looks good to me as a place to start

File: interfaces/lego-boost-sse/README.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+* A Lego Boost robot running on 2 wheels (either tank or regular)
+* A laptop to be able to run the Vuforia Spatial Edge Server
+* A mobile phone with the Vuforia Spatial Toolbox application installed
+* You will also need the [Vuforia](https://developer.vuforia.com/) object target or image target to identify the robot with the mobile device [Later explained]
+
+### Download code
+
+1. Download the [vuforia-spatial-edge-server](https://github.com/ptcrealitylab/vuforia-spatial-edge-server) code and place it in your Development folder on your computer.
+
+2. Download the [vuforia-spatial-robotic-addon](https://github.com/ptcrealitylab/vuforia-spatial-robotic-addon) code and place it in the following folder:
```
Comment (benptc) [inline interfaces/lego-boost-sse/README.md:38]: Do all the references to "vuforia-spatial-robotic-addon" need to be updated to "vuforia-spatial-motion-addon" now that we've moved this to a different repo?

File: interfaces/lego-boost-sse/index.js [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+            pathData.push({
+                id: pathPointId,
+                objectId: pathPointObjectId,
+                speed: point.speed,
+                x: point.matrix[12] / point.matrix[15],
+                y: (-1)*point.matrix[13] / point.matrix[15],
+                z: point.matrix[14] / point.matrix[15]
+            });
+        });
+    });
```
Comment (benptc) [inline interfaces/lego-boost-sse/index.js:319]: Looks good!

File: interfaces/lego-boost-sse/tools/mission/index.html (hunk 1 above)
Comment (benptc) [reply interfaces/lego-boost-sse/tools/mission/index.html:1]: We have an API now in hardwareInterfaces.js called "setTool" that allows a hardware interface to instantiate a tool on an object if it's stored in this location. I think Valentin added this feature.

**Review by benptc — APPROVED** [review]
Overall looks good! Left a comment about the README but otherwise I think it's a good first version.

File: interfaces/lego-boost-sse/README.md (hunk 2 above)
Comment (afustePTC) [reply interfaces/lego-boost-sse/README.md:38]: Yep. will update all of this and create a new readme in the root

File: interfaces/lego-boost-sse/tools/mission/index.html (hunk 1 above)
Comment (afustePTC) [reply interfaces/lego-boost-sse/tools/mission/index.html:1]: It is meant to have custom tools for specific hardware interfaces

**Pull request merged**